components/service-operator/controllers/postgres_cloudformation_test.go

```diff
@@ -0,0 +1,163 @@
+				Namespace: namespace,
+				Name:      name,
+			}
+			secretNamespacedName = types.NamespacedName{
+				Namespace: namespace,
+				Name:      secretName,
+			}
+			pg = database.Postgres{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+					Labels: map[string]string{
+						cloudformation.AccessGroupLabel: "test.access.group",
+					},
+				},
+				Spec: database.PostgresSpec{
+					Secret: secretName,
+					AWS: database.PostgresAWSSpec{
+						InstanceCount: 1,
+						InstanceType:  "db.t3.medium",
+					},
+				},
+			}
+			secret core.Secret
+		)
+
+		By("creating an resource with kubernetes api", func() {
+			Expect(client.Create(ctx, &pg)).To(Succeed())
+		})
+
+		By("displaying a READY resource status after initial creation", func() {
+			Eventually(func() object.State {
+				_ = client.Get(ctx, resourceNamespacedName, &pg)
+				return pg.GetState()
+			}, timeout).Should(Equal(object.ReadyState))
+		})
+
+		By("displaying an AWS CREATE_COMPLETE resource status after initial creation", func() {
+			Eventually(func() string {
+				_ = client.Get(ctx, resourceNamespacedName, &pg)
+				return pg.Status.AWS.Status
+			}, timeout).Should(Equal(cloudformation.CreateComplete))
+		})
+
+		By("displaying an AWS stack id in resource status", func() {
+			Eventually(func() string {
+				_ = client.Get(ctx, resourceNamespacedName, &pg)
+				return pg.Status.AWS.ID
+			}).ShouldNot(BeEmpty())
+		})
+
+		By("displaying a stack name prefixed with cluster name in resource status", func() {
+			Eventually(func() string {
+				_ = client.Get(ctx, resourceNamespacedName, &pg)
+				return pg.Status.AWS.Name
+			}).Should(ContainSubstring("xxx-postgres-test-test-db"))
+		})
+
+		By("ensuring a finalizaer is present on resource to prevent deletion", func() { // TODO: move to cloudformation.Controller unit test
+			Eventually(func() []string {
+				_ = client.Get(ctx, resourceNamespacedName, &pg)
+				return pg.Finalizers
+			}).Should(ContainElement(cloudformation.Finalizer))
+		})
+
+		By("ensuring no DeletionTimestamp exists", func() { // TODO: move to cloudformation.Controller unit test
+			Eventually(func() bool {
+				_ = client.Get(ctx, resourceNamespacedName, &pg)
+				return pg.ObjectMeta.DeletionTimestamp == nil
+			}).Should(BeTrue())
+		})
+
+		By("creating a secret with username", func() {
+			Eventually(func() map[string][]byte {
+				_ = client.Get(ctx, secretNamespacedName, &secret)
+				return secret.Data
+			}).Should(HaveKey("Username"))
+		})
+
+		By("creating a secret with password", func() {
+			Eventually(func() map[string][]byte {
+				_ = client.Get(ctx, secretNamespacedName, &secret)
+				return secret.Data
+			}).Should(HaveKey("Password"))
+		})
+
+		By("creating a secret with endpoint", func() {
+			Eventually(func() map[string][]byte {
+				_ = client.Get(ctx, secretNamespacedName, &secret)
+				return secret.Data
+			}).Should(HaveKey("Endpoint"))
```

Comment:
Also we probably want one for ReadEndpoint

Reply:
Gonna add these into unit tests for GetStackTemplate on the types in future PR